quota-notification/src/main/java/functions/ConfigureAppAlertHelper.java

```diff
@@ -0,0 +1,524 @@
+import com.google.cloud.bigquery.TableResult;
+import com.google.cloud.logging.Logging;
+import com.google.cloud.logging.LoggingOptions;
+import com.google.cloud.logging.MetricInfo;
+import com.google.cloud.monitoring.v3.AlertPolicyServiceClient;
+import com.google.cloud.monitoring.v3.NotificationChannelServiceClient;
+import com.google.cloud.storage.Blob;
+import com.google.cloud.storage.Storage;
+import com.google.cloud.storage.StorageOptions;
+import com.google.monitoring.v3.Aggregation;
+import com.google.monitoring.v3.Aggregation.Aligner;
+import com.google.monitoring.v3.AlertPolicy;
+import com.google.monitoring.v3.AlertPolicy.Condition;
+import com.google.monitoring.v3.AlertPolicy.Condition.MetricThreshold;
+import com.google.monitoring.v3.AlertPolicy.Condition.Trigger;
+import com.google.monitoring.v3.AlertPolicy.ConditionCombinerType;
+import com.google.monitoring.v3.AlertPolicy.Documentation;
+import com.google.monitoring.v3.ComparisonType;
+import com.google.monitoring.v3.NotificationChannel;
+import com.google.monitoring.v3.UpdateAlertPolicyRequest;
+import com.google.monitoring.v3.UpdateNotificationChannelRequest;
+import com.google.protobuf.Duration;
+import functions.eventpojos.AppAlert;
+import java.io.BufferedReader;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import com.google.monitoring.v3.ProjectName;
+import java.nio.channels.Channels;
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.UUID;
+import java.util.logging.Logger;
+import org.apache.commons.lang3.StringUtils;
+
+
+public class ConfigureAppAlertHelper {
+  private static final String HOME_PROJECT = ConfigureAppAlert.HOME_PROJECT;
+  private static final String DATASET = ConfigureAppAlert.APP_ALERT_DATASET;
+  private static final String TABLE = ConfigureAppAlert.APP_ALERT_TABLE;
+  private static final String CSV_SOURCE_URI = ConfigureAppAlert.CSV_SOURCE_URI;
+
+  private static final Logger logger = Logger.getLogger(ConfigureAppAlertHelper.class.getName());
+
+
+
+  /*
+   * API to initialize App Alert table.
+   * This API loads data from csv file in Cloud Storage bucket to BigQuery table
+   * */
+  public static void loadCsvFromGcsToBigQuery(BigQuery bigquery, TableId tableId){
+    // Do not upload data from csv if table already contains data
+    // This batch upload is used only for initializing the data for the first time
+    // for subsequent addition/modification of data, use DML queries in BigQuery table
+    if(!isAppAlertTableEmpty(bigquery) )
+      return;
+    // Parse the csv file and verify that
+    // 1. File is available,
+    // 2. There are no records with empty/null project_id, email_id or app_code
+    // 3. There are no duplicates for app_code
+    if(isCSVParseSuccess()){
+      logger.info("CSV parsed successfully! Loading data in BigQuery...");
+    } else {
+      logger.severe("CSV parsing failed. Fix the CSV to proceed.");
```

Comment:
Does the severe level equate to something that causes the function to a failed run, such that it shows up in the Cloud Functions UI as a failure, so users are more proactively notified?